extensions/wikia/ArticleVideo/bower_components/html5-skin/js/components/controlBar.js

```diff
@@ -0,0 +1,506 @@
+  },
+
+  handleShareClick: function() {
+    this.props.controller.toggleShareScreen();
+  },
+
+  handleQualityClick: function() {
+    if(this.props.responsiveView == this.props.skinConfig.responsive.breakpoints.xs.id) {
+      this.props.controller.toggleScreen(CONSTANTS.SCREEN.VIDEO_QUALITY_SCREEN);
+    } else {
+      this.toggleQualityPopover();
+      this.closeCaptionPopover();
+    }
+  },
+
+  toggleQualityPopover: function() {
+    this.props.controller.toggleVideoQualityPopOver();
+  },
+
+  closeQualityPopover: function() {
+    if(this.props.controller.state.videoQualityOptions.showVideoQualityPopover == true) {
+      this.toggleQualityPopover();
+    }
+  },
+
+  toggleCaptionPopover: function() {
+    this.props.controller.toggleClosedCaptionPopOver();
+  },
+
+  closeCaptionPopover: function() {
+    if(this.props.controller.state.closedCaptionOptions.showClosedCaptionPopover == true) {
+      this.toggleCaptionPopover();
+    }
+  },
+
+  closePopovers: function() {
+    this.closeCaptionPopover();
+    this.closeQualityPopover();
+  },
+
+  handleVolumeClick: function(evt) {
+    evt.preventDefault();
+    var newVolume = parseFloat(evt.target.dataset.volume);
+    this.props.controller.setVolume(newVolume);
+  },
+
+  handleDiscoveryClick: function() {
+    this.props.controller.toggleDiscoveryScreen();
+  },
+
+  handleMoreOptionsClick: function() {
+    this.props.controller.toggleMoreOptionsScreen(this.moreOptionsItems);
+  },
+
+  handleClosedCaptionClick: function() {
+    if(this.props.responsiveView == this.props.skinConfig.responsive.breakpoints.xs.id) {
+      this.props.controller.toggleScreen(CONSTANTS.SCREEN.CLOSEDCAPTION_SCREEN);
+    } else {
+      this.toggleCaptionPopover();
+      this.closeQualityPopover();
+    }
+  },
+
+  //TODO(dustin) revisit this, doesn't feel like the "react" way to do this.
+  highlight: function(evt) {
+    var color = this.props.skinConfig.controlBar.iconStyle.active.color ? this.props.skinConfig.controlBar.iconStyle.active.color : this.props.skinConfig.general.accentColor;
+    var opacity = this.props.skinConfig.controlBar.iconStyle.active.opacity;
+    Utils.highlight(evt.target, opacity, color);
+  },
+
+  removeHighlight: function(evt) {
+    var color = this.props.skinConfig.controlBar.iconStyle.inactive.color;
+    var opacity = this.props.skinConfig.controlBar.iconStyle.inactive.opacity;
+    Utils.removeHighlight(evt.target, opacity, color);
+  },
+
+  volumeHighlight:function() {
+    this.highlight({target: ReactDOM.findDOMNode(this.refs.volumeIcon)});
+  },
+
+  volumeRemoveHighlight:function() {
+    this.removeHighlight({target: ReactDOM.findDOMNode(this.refs.volumeIcon)});
+  },
+
+  changeVolumeSlider: function(event) {
+    var newVolume = parseFloat(event.target.value);
+    this.props.controller.setVolume(newVolume);
+    this.setState({
+      volumeSliderValue: event.target.value
+    });
+  },
+
+  populateControlBar: function() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Function has a complexity of 50 which is greater than 10 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)